ansible/plugins/connection/switch.py

```diff
@@ -231,8 +235,9 @@ def exec_command(self, *args, **kwargs):
             self._display.vvv('> %s' % (cmd), host=self.host)
             client.sendline(cmd)
             client.expect(prompts)
-            stdout += client.before
-            self._display.vvv('< %s' % (client.before), host=self.host)
+            before = self._remove_unprintable(client.before)
+            stdout += before
+            self._display.vvv('< %s' % (before), host=self.host)
```

Comment:
I believe this approach can avoid decoding the issue. But potentially some useful information may be filtered out from the output. Could you try this approach to see if it works?:
```
before = client.before.encode('ascii', 'backslashreplace')
stdout += before
self._display.vvv('< %s' % (before), host=self.host)
```

Reply:
If to check ASCII table, it looks like that in our case, symbols > 128 will not be from useful information.
I would prefer not to save such characters to avoid further issues in string buffer manipulations.

What do you think?